tests/beacon/conftest.py

```diff
@@ -0,0 +1,221 @@
+    return list(keymap)
+
+
+@pytest.fixture
+def sample_active_state_params():
+    return {
+        'pending_attestations': [],
+        'recent_block_hashes': [],
+    }
+
+
+@pytest.fixture
+def sample_attestation_record_params():
+    return {
+        'slot': 10,
+        'shard_id': 12,
+        'oblique_parent_hashes': [],
+        'shard_block_hash': b'\x20' * 32,
+        'attester_bitfield': b'\x33\x1F',
+        'justified_slot': 5,
+        'justified_block_hash': b'\x33' * 32,
+        'aggregate_sig': [0, 0],
+    }
+
+
+@pytest.fixture
+def sample_block_params():
+    return {
+        'parent_hash': b'\x55' * 32,
+        'slot_number': 10,
+        'randao_reveal': b'\x34' * 32,
+        'attestations': [],
+        'pow_chain_ref': b'\x32' * 32,
+        'active_state_root': b'\x01' * 32,
+        'crystallized_state_root': b'\x05' * 32
+    }
+
+
+@pytest.fixture
+def sample_crystallized_state_params():
+    return {
+        'validators': [],
+        'last_state_recalc': 50,
+        'shard_and_committee_for_slots': [],
+        'last_justified_slot': 100,
+        'justified_streak': 10,
+        'last_finalized_slot': 70,
+        'current_dynasty': 4,
+        'crosslink_records': [],
+        'dynasty_seed': b'\x55' * 32,
+        'dynasty_start': 3,
+    }
+
+
+@pytest.fixture
+def sample_crosslink_record_params():
+    return {
+        'dynasty': 2,
+        'slot': 0,
+        'hash': b'\x43' * 32,
+    }
+
+
+@pytest.fixture
+def sample_recent_proposer_record_params():
+    return {
+        'index': 10,
+        'randao_commitment': b'\x43' * 32,
+        'balance_delta': 3
+    }
+
+
+@pytest.fixture
+def sample_shard_and_committee_params():
+    return {
+        'shard_id': 10,
+        'committee': [],
+    }
+
+
+@pytest.fixture
+def sample_validator_record_params():
+    return {
+        'pubkey': 123,
+        'withdrawal_shard': 10,
+        'withdrawal_address': b'\x01' * 20,
+        'randao_commitment': b'\x01' * 32,
+        'balance': 100,
+        'start_dynasty': 1,
+        'end_dynasty': 3
+    }
+
+
+#
+# config
+#
+
+@pytest.fixture
+def init_shuffling_seed():
+    return DEFAULT_SHUFFLING_SEED
+
+
+@pytest.fixture
+def init_randao():
+    return DEFAULT_RANDAO
+
+
+@pytest.fixture
+def base_reward_quotient():
+    return BASE_REWARD_QUOTIENT
+
+
+@pytest.fixture
+def default_end_dynasty():
+    return DEFAULT_END_DYNASTY
+
+
+@pytest.fixture
+def deposit_size():
+    return DEPOSIT_SIZE
+
+
+@pytest.fixture
+def cycle_length():
+    return CYCLE_LENGTH
+
+
+@pytest.fixture
+def max_validator_count():
+    return MAX_VALIDATOR_COUNT
+
+
+@pytest.fixture
+def min_committee_size():
+    return MIN_COMMITTEE_SIZE
+
+
+@pytest.fixture
+def min_dynasty_length():
+    return MIN_DYNASTY_LENGTH
+
+
+@pytest.fixture
+def shard_count():
+    return SHARD_COUNT
+
+
+@pytest.fixture
+def slot_duration():
+    return SLOT_DURATION
+
+
+@pytest.fixture
+def sqrt_e_drop_time():
+    return SQRT_E_DROP_TIME
```

Comment:
This seems like a lot of code overhead for a bunch of things that are simple defaults.  We could drop all of these fixtures and just bake these defaults into the `config` fixture.

Reply:
I personally like this construction because I can do things like the following:

```python
@pytest.mark.parametrize(
    (
        'cycle_length,min_committee_size,shard_count'
    ),
    [
        (50, 10, 10),
        (20, 10, 100),
    ],
)
def test(genesis_crystallized_state,
         genesis_active_state,
         mock_make_block):
    do_test()
```
This allows for setting these configuration variables via `parametrize` and the config vars that are set this way percolate through the fixtures, whereas the ones untouched stay as defaults.

I agree it's a strange construction. Do you have a better alternative?

Reply:
Maybe something with environment variables and potentially the `monkeypatch` fixture, though I'm moving more in favor of removing the `config` *global* concept towards a `constants` module and configurable class based approach.

I.E. The `cycle_length` would be part of some class and retrieved with `BeaconChain.get_cycle_length()` with the default implementation returning `constants.CYCLE_LENGTH`.

- Easy to modify at runtime
- easy to integrate into the chain builder API
- avoids the use of globals
- doesn't require passing around a config object (which will result in annoying dependencies based on my experience

Reply:
Oh, and if other agree to go with this approach I would suggest not doing it in this PR, but rather getting this cleaned up, merging, and doing it in a subsequent one.

Reply:
sounds good 👍